README.md

```diff
@@ -1,84 +1,59 @@
+- Home Formats: returns an array of available home formats. `["DVD", "BD"]`, for example.
+
+The API Search gem will give you the following search functionality:
+
+Query String Example  | What Data is Returned
+------------- | -------------
+`movies` | All data in your data set
+`movies?q=rating=all` | Movies with a non-nil rating
+`movies?q=rating=none` | Movies with a nil rating
+`movies?q=year=1994` | Movies with a year of 1994
+`movies?q=year=1994|1995` | Movies with a year of 1994 or 1995
+`movies?q=title=Dunston%20Checks%20In` | Movies with the title Dunston Checks In
+`movies?q=star_rating>1` | Movies with a star rating greater than one
+`movies?q=star_rating<3` | Movies with a star rating less than three
+`movies?q=star_rating>=2` | Movies with a star rating greater than or equal to 2
+`movies?q=star_rating<=4` | Movies with a star rating less than or equal to 4
+`movies?q=home_formats=DVD` | Movies that are available on DVD.
+`movies?q=home_formats=DVD|BD` | Movies that are available on DVD or Blu Ray
 
 ### Combining Searches
 
-Search criteria can be separated with commas
+Any of the above critera can be combined. 
+Just separate criteria with commas. 
+Records that match **all** the criteria will be returned. 
+Some examples!
 
-`movie?q=year=1994,country=US,rated=none`
+`movie?q=year=1994,country=US,rated=none`<br />
+ All un-rated movies made in the US in 1994.
 
-Records that match **all** the criteria will be returned. All un-rated movies made in the US in 1994.
+`movie?q=home_formats=BD|DVD,rated=R|PG,star_rating>3`<br />
+All movies available on BD or DVD, that are rated R or PG, and that have a star rating greater than 3
 
-## Installation
+## Use Cases
 
-Add this line to your application's Gemfile:
+This gem fits best when you're trying to provide very flexible searching on a relatively small and flat set of data. Because it filters a data set that you provide, the speed at which it works is wholly dependent on the size of your data set.
 
-```ruby
-gem 'query_string_search'
-```
+We currently use this gem on a small (~2000 rows) data set which is cached in memory. Response times using Rails/Passenger/Apache are always under 100ms, usually under 50ms. That's plenty fast! But if your initial data set is massive, or you can't cache it, your response times will be slower. Such is life. Gems like [Periscope](https://rubygems.org/gems/periscope) or [has_scope](https://rubygems.org/gems/has_scope) may be more your speed.
 
-And then execute:
+A downside of gems like those, though, is that you end up defining all of your filtering methods in advance. This is not the case with Query String Search. If the elements in your data set respond to a method, then the data can be filtered by that attribute. It is possible that this is not what you want. In which case, check out those other gems. Or, you could also wrap your object instances in a wrapper that only responds to the methods that you want to filter on.
```

Comment:
We talk about filtering based on elements' methods and then talk about attributes: "If the elements in your data set respond to a method, then the data can be filtered by that attribute." 

While these are one and the same, it reads a little awkward. 

How about: 
"If the elements in your data set respond to a method (for example an attribute), then the data can be filtered by the value of that method." 
Or 
"Your data set can be filtered by any attribute (or method) your elements respond to"